tests/ImageSharp.Tests/PixelFormats/PixelOperations/Generated/_Common.ttinclude

```diff
@@ -0,0 +1,119 @@
+<#@ template debug="false" hostspecific="false" language="C#" #>
+<#@ assembly name="System.Core" #>
+<#@ import namespace="System.Linq" #>
+<#@ import namespace="System.Text" #>
+<#@ import namespace="System.Collections.Generic" #>
+// Copyright (c) Six Labors.
+// Licensed under the Apache License, Version 2.0.
+
+// <auto-generated />
+
+using System.Runtime.CompilerServices;
+using SixLabors.ImageSharp.PixelFormats;
+using Xunit;
+using Xunit.Abstractions;
+<#+
+    private static readonly string[] UnassociatedAlphaPixelTypes =
+    {
+        "A8",
+        "Argb32",
+        "Bgra32",
+        "Bgra4444",
+        "Bgra5551",
+        "Byte4",
+        "HalfVector4",
+        "La16",
+        "La32",
+        "NormalizedByte4",
+        "NormalizedShort4",
+        "Rgba1010102",
+        "Rgba32",
+        "Rgba64",
+        "RgbaVector",
+        "Short4"
+    };
+
+    private static readonly string[] AssociatedAlphaPixelTypes = Array.Empty<string>();
+
+    private static readonly string[] CommonPixelTypes =
+    {
+        "A8",
+        "Argb32",
+        "Bgr24",
+        "Bgr565",
+        "Bgra32",
+        "Bgra4444",
+        "Bgra5551",
+        "Byte4",
+        "HalfSingle",
+        "HalfVector2",
+        "HalfVector4",
+        "L16",
+        "L8",
+        "La16",
+        "La32",
+        "NormalizedByte2",
+        "NormalizedByte4",
+        "NormalizedShort2",
+        "NormalizedShort4",
+        "Rg32",
+        "Rgb24",
+        "Rgb48",
+        "Rgba1010102",
+        "Rgba32",
+        "Rgba64",
+        "RgbaVector",
+        "Short2",
+        "Short4",
+    };
+
+    void GenerateSpecializedClass(string pixelType, string alpha)
+    {
+#>
+        public partial class <#=pixelType#>_OperationsTests : PixelOperationsTests<<#=pixelType#>>
+        {
+            public <#=pixelType#>_OperationsTests(ITestOutputHelper output)
+                : base(output)
+            {
+                var alphaRepresentation =  <#=pixelType#>.PixelOperations.Instance.GetPixelTypeInfo().AlphaRepresentation;
+                this.HasUnassociatedAlpha = alphaRepresentation == PixelAlphaRepresentation.Unassociated;
+            }
+
+            [Fact]
+            public void IsSpecialImplementation() => Assert.IsType<<#=pixelType#>.PixelOperations>(PixelOperations<<#=pixelType#>>.Instance);
+
+            [Fact]
+            public void PixelTypeInfoHasCorrectBitsPerPixel()
+            {
+                var bits = <#=pixelType#>.PixelOperations.Instance.GetPixelTypeInfo().BitsPerPixel;
+                Assert.Equal(Unsafe.SizeOf<<#=pixelType#>>() * 8, bits);
+            }
+
+            [Fact]
+            public void PixelTypeInfoHasCorrectAlphaRepresentation()
+            {
+                var alphaRepresentation = <#=pixelType#>.PixelOperations.Instance.GetPixelTypeInfo().AlphaRepresentation;
+                Assert.Equal(<#=alpha#>, alphaRepresentation);
```

Comment:
I'm concerned whether this type of testing is valuable here.

I would rather try something like this in the base type:

```C#
[Fact]
public void PixelAlphaRepresentation_DefinesPresenceOfAlphaChannel()
{
    TPixel pixel = default;
    pixel.FromRgba32(new Rgba32(0, 0, 0, 123));

    Rgba32 dest;
    pixel.ToRgba32(ref dest);

    bool hasAlpha = GetPixelTypeInfo().AlphaRepresentation != PixelAlphaRepresentation.None;
    byte exectedAlpha = hasAlpha ? 123 : 255;    

    Assert.Equal(expectedAlpha, dest.A);
}
```

Reply:
All I'm testing here is the correct assignment of the value. Any behavior based upon that value is the responsibility of the caller.

Reply:
That test could be useful though to check I assigned it correctly.

Reply:
> All I'm testing here is the correct assignment of the value.

My point explained further:
it's very easy to make a mistake in such a test here (tempting to copy-paste product code to test code brainlessly, especially if it's generated), but practically impossible, if the test is validating against actual behavior.

But I'm fine with keeping this as an improvement opportunity for later.

Reply:
I've implemented a version of the test in the base class. The original test remains in the generated versions though as I do not use code generation for this property in the source.